docs/topics/object-declarations.md

```diff
@@ -294,65 +233,336 @@ class MyClass {
+    fun getNotification() = object {
+        val message: String = "General notification"
+    }
+
+    // The return type is Notification because the anonymous object implements only one interface
+    // The notifyUser() function is accessible because it is part of the Notification interface
+    // The message property is not accessible because it is not declared in the Notification interface
+    fun getEmailNotification() = object : Notification {
+        override fun notifyUser() {
+            println("Sending email notification")
+        }
+        val message: String = "You've got mail!"
+    }
+
+    // The return type is DetailedNotification. The notifyUser() function and the message property are not accessible
+    // Only members declared in the DetailedNotification interface are accessible
+    fun getDetailedNotification(): DetailedNotification = object : Notification, DetailedNotification {
+        override fun notifyUser() {
+            println("Sending detailed notification")
+        }
+        val message: String = "Detailed message content"
+    }
+}
+//sampleEnd
+fun main() {
+    // This produces no output
+    val notificationManager = NotificationManager()
+
+    // The message property is not accessible here because the return type is Any
+    // This produces no output
+    val notification = notificationManager.getNotification()
+
+    // The notifyUser() function is accessible
+    // The message property is not accessible here because the return type is Notification
+    val emailNotification = notificationManager.getEmailNotification()
+    emailNotification.notifyUser()
+    // Sending email notification
+
+    // The notifyUser() function and message property are not accessible here because the return type is DetailedNotification
+    // This produces no output
+    val detailedNotification = notificationManager.getDetailedNotification()
+}
+```
+{kotlin-runnable="true" id="object-expression-object-override"}
+
+### Access variables from anonymous objects
+
+Code within the body of object expressions can access variables from the enclosing scope:
+
+```kotlin
+import java.awt.event.MouseAdapter
+import java.awt.event.MouseEvent
+
+fun countClicks(window: JComponent) {
+    var clickCount = 0
+    var enterCount = 0
+
+    // MouseAdapter provides default implementations for mouse event functions
+    // Simulates MouseAdapter handling mouse events
+    window.addMouseListener(object : MouseAdapter() {
+        override fun mouseClicked(e: MouseEvent) {
+            clickCount++
+        }
+
+        override fun mouseEntered(e: MouseEvent) {
+            enterCount++
+        }
+    })
+    // The clickCount and enterCount variables are accessible within the object expression
+}
+```
+
+## Behavior difference between object expressions and declarations
+
+There are differences in the initialization behavior between object expressions and object declarations:
```

Comment:
```suggestion
There are differences in the initialization behavior between object declarations and object expressions:
```
For symmetry. Might be worth swapping the order of the first two bullets as well. I leave it up to you though!